src/schedule/edf_schedule.c

```diff
@@ -267,15 +273,18 @@ int scheduler_init_edf(void)
 	task_main_init();
 
 	/* configure EDF scheduler interrupt */
-	edf_sch->irq = interrupt_get_irq(PLATFORM_SCHEDULE_IRQ,
-					 PLATFORM_SCHEDULE_IRQ_NAME);
-	if (edf_sch->irq < 0)
+	ret = interrupt_get_irq(PLATFORM_SCHEDULE_IRQ,
+				PLATFORM_SCHEDULE_IRQ_NAME);
+	if (edf_sch->irq < 0) {
```

Comment:
something's wrong here. You probably don't need `ret`, just keep the original assignment

Reply:
fixed the ret check, but we cant use edf_sch->irq for return value since we have to now free edf_sch.